### PR TITLE
Release 8.4.0: New variable 'cpu_credits' in asg and asg-ebs modules

### PR DESCRIPTION
**Pull request by dalenewby**
## Added

- Added optional variable `cpu_credits` to the `asg` and `asg-ebs` modules.

### CODE REVIEW

**Review by briskt — DISMISSED** [review]

**Comment by briskt** [comment]
We often put "Release" and the expected release number at the beginning of the pr title. (E.g. "Release 1.2.3 - fixed some stuff") Just helps to highlight the fact that it's a release request.

**Review by briskt — DISMISSED** [review]

File: test/asg.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module "asg-test" {
+  source                  = "../aws/asg"
+  app_name                = "test_name1"
+  app_env                 = "test_env1"
+  ami_id                  = "ami-1234567890"
+  aws_instance            = { instance_type = "t3.nano", volume_size = "8", instance_count = "1" }
+  private_subnet_ids      = ["1234", "5678"]
+  default_sg_id           = "sg-asdf"
+  ecs_instance_profile_id = "asdf"
+  ecs_cluster_name        = "mycluster1"
```
Comment (Baggerone) [inline test/asg.tf:11]: Do you want to include `cpu_credits` in one of these?

**Review by Baggerone — DISMISSED** [review]

File: test/asg.tf (hunk 1 above)
Comment (dalenewby) [reply test/asg.tf:11]: I assumed we were only including the *required* inputs.  Is that a bad assumption?  Should we include the other optional inputs as well?  If some are mutually exclusive, do we create separate tests for the valid combinations?

Comment (Baggerone) [reply test/asg.tf:11]: @briskt Do you have an opinion on this?

Comment (briskt) [reply test/asg.tf:11]: Not a strong opinion. I've done it a couple ways. Minimum set of parameters, with only the required ones. Or all parameters. Or one of each

**Review by Baggerone — APPROVED** [review]

**Pull request merged**